pkg/restapi/operation/didexchange/didexchange_test.go

```diff
@@ -319,9 +318,28 @@ func TestServiceEvents(t *testing.T) {
 	didExSvc, err := didexsvc.New(&did.MockDIDCreator{}, &protocol.MockProvider{CustomStore: store})
 	require.NoError(t, err)
 
+	done := make(chan struct{})
+
 	// create the client
 	op, err := New(&mockprovider.Provider{StorageProviderValue: mockstore.NewMockStoreProvider(),
-		ServiceValue: didExSvc}, webhook.NewHTTPNotifier(nil))
+		ServiceValue: didExSvc},
+		&mockNotifier{
+			notifyFunc: func(topic string, message []byte) error {
+				require.Equal(t, "connections", topic)
+
+				conn := didexchange.Connection{}
+				jsonErr := json.Unmarshal(message, &conn)
+				require.NoError(t, jsonErr)
+
+				if conn.State == "responded" {
```

Comment:
Perhaps these connection states should be constants defined somewhere as well, since I think there's a pre-defined finite number of them right?

Reply:
This is a webhook consumer validation (a separate process). So i feel it would make sense to have a validation against a literal rather than against a lib constant.

Reply:
True, but it does use the same restAPI + webhook library, so it doesn't look totally separate to me code wise...

Reply:
Webhook server won't use aries library. it'll be a separate server/process. Webhook server and Agent communicate through RestAPIs.

Reply:
Hmm... is there some value to not extracting that string out into the payloads.go file? Or some other common file? If the names are ever changed in the future then we'll have to update in multiple places.

Reply:
I understand what you are trying to say. The State names are [internal](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/protocol/didexchange/states.go#L32). And here the literal is defined only once (so it won't make much sense to move it to constant).

Reply:
Gotcha. Makes sense!